cogs/events.py

```diff
@@ -1,148 +1,132 @@
 
         # Check if user desires to have something other than online
-        status_type = {
+        from_status_type = {
+            "online": discord.Status.online,
             "idle": discord.Status.idle,
-            "dnd": discord.Status.dnd
+            "dnd": discord.Status.dnd,
+            "offline": discord.Status.invisible,
+            "invisible": discord.Status.invisible,
         }
 
         # Check if user desires to have a different type of activity
-        activity_type = {
+        from_activity_type = {
             "listening": discord.ActivityType.listening,
             "watching": discord.ActivityType.watching,
             "streaming": discord.ActivityType.streaming,
-            "competing": discord.ActivityType.competing
+            "competing": discord.ActivityType.competing,
+            "custom": discord.ActivityType.custom,
         }
 
-        activity_url = self.bot.config.discord_status_url
-        guild_id = self.bot.config.discord_guild_id
-
-        await self.bot.change_presence(
-            activity=discord.Activity(
-                type=activity_type.get(
-                    self.bot.config.discord_activity_type.lower(),
-                    discord.ActivityType.playing
-                ),
-                name=self.bot.config.discord_activity_name,
-                url=(
-                    f"https://twitch.tv/{activity_url}"
-                    if activity_url else None
-                ),
-            ),
-            status=status_type.get(
-                self.bot.config.discord_status_type.lower(),
-                discord.Status.online
-            )
-        )
+        activity_type = self.bot.config.discord_activity_type.lower()
+        guild_id = self.bot.config.owner_guild_id
+        if activity_type not in from_activity_type:
+            raise ValueError(f"Activity type must be one of {', '.join(from_activity_type.keys())}")
 
-        await self.bot.tree.sync(
-            guild=(
-                discord.Object(id=guild_id)
-                if guild_id else None
-            )
+        activity_url = self.bot.config.streaming_status_url
+        if activity_type == "streaming" and not activity_url:
+            raise ValueError("Streaming status URL must be set in .env to use the streaming activity type.")
+
+        status_type = self.bot.config.discord_status_type.lower()
+        if status_type not in from_status_type:
+            raise ValueError(f"Status type must be one of {', '.join(from_status_type.keys())}")
+
+        activity = discord.Activity(
+            name=self.bot.config.discord_activity_name,
+            type=from_activity_type[activity_type],
+            url=activity_url,
+            state=self.bot.config.discord_activity_name if activity_type == "custom" else None,
         )
 
+        await self.bot.change_presence(status=from_status_type[status_type], activity=activity)
+
+        global_synced = await self.bot.tree.sync()
```

Comment:
Syncing app commands on startup is also not needed. This is especially bad in on_ready as it can be fired multiple times. Do not sync in on_ready.